guide_service_errors.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: guide
+title: Configuring Service Errors
+permalink: /user_guide/service_errors
+index: 2
+excerpt: This page describes how to configure service errors for a resource.
+---
+
+# Configuring Service Errors
+
+{{ page.excerpt }}
+
+## Contents
+
+-   [Background](#background)
+-   [Step 1: Define your Service Errors](#step-1-define-your-service-errors)
+-   [Step 2: Write Your Service Error Definition](#step-2-write-your-service-error-definition)
+-   [Step 3: Apply the Service Errors to your Resource](#step-3-apply-the-service-errors-to-your-resource)
+-   [Step 4: Returning Service Errors](#step-4-returning-service-errors)
+-   [Step 5: Enabling Service Error Validation](#step-5-enabling-service-error-validation)
+
+## Background
+
+It's important to communicate to the consumers of a service what types of errors
+may be returned by a particular API. Fortunately, Rest.li allows application developers to
+configure service errors for their resources. This makes APIs clearer by documenting in
+the IDL what sorts of errors should be expected, allowing clients to effectively handle
+and process returned errors. Also, enabling service error validation allows services
+to guarantee which errors a client should expect to encounter.
+
+## Step 1: Define your Service Errors
+
+First, you need to determine which service error codes your service may return, and what
+information will be contained in error responses corresponding with these codes. You'll
+need to define four things:
+
+- The application-specific string code associated with some service error
```

Comment:
HTTP status text?

Reply:
No, this refers to something like `"BAD_REQUEST"`, I can add this as an example.

Reply:
`BAD_REQUEST` seems the HTTP status text. HTTP status has a code and a text. For example, `404` -> `NOT_FOUND` ....

Reply:
That was a bad example because it is also an HTTP status text. I mean something application-specified like `INVALID_ALBUM_ID` or `QUOTA_EXCEEDED`